CMakeLists.txt

```diff
@@ -336,12 +336,16 @@ if (CLR_CMAKE_PLATFORM_UNIX)
   add_definitions(-DPLATFORM_UNIX=1)
   add_definitions(-DFEATURE_PAL_SXS)
 
-  if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+  # CMAKE_SYSTEM_PROCESSOR returns the value of `uname -p`.
+  # For the AMD/Intel 64bit architecure two different strings are common.
+  # Linux and Darwin identify it as "x86_64" while FreeBSD uses the
+  # "amd64" string. Accept either of the two here.
```

Comment:
Could you please move this comment to the first place where the long condition is checked?